enhancements/oc/oc-login-external-oidc-integration.md

```diff
@@ -0,0 +1,238 @@
+      command: oc
+      env: null
+      installHint: Please be sure that oc is defined in $PATH to be executed as credentials
+        exec plugin
+      interactiveMode: IfAvailable
+      provideClusterInfo: false
+```
+
+Moreover, `oc login` command triggers authentication process by sending a request to Project API's whoami endpoint automatically so that
+client-go talks to `oc get-token` command to obtain id token (and also refresh token, if provider supports it).
+
+There are currently 7 authentication flows grouped around public and confidential clients. Supportability (in terms of authentication flows) matrix of this feature is outlined below; 
+
+|                  | Authorization Code<br/>(public)                                            | Authorization Code<br/>(confidential)                                                                    | Authorization Code + PKCE<br/>(public)                                     | Authorization Code + PKCE<br/>(confidential)                                                             | Implicit Flow<br/>(public) | Client Credentials<br/>(confidential)                     | Device Code<br/>(public)                                  | Password Grant<br/>(confidential) |
+|------------------|----------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|----------------------------|-----------------------------------------------------------|-----------------------------------------------------------|-----------------------------------|
+| Supported        | Yes                                                                        | Yes                                                                                                      | Yes                                                                        | Yes                                                                                                      | No                         | No                                                        | No                                                        | No                                |
+| Comment          | Keycloak - [configurable] Azure Entra ID - [configurable]                  | Keycloak - [not-configurable] Azure Entra ID - [configurable]                                            | Keycloak - [configurable] Azure Entra ID - [configurable]                  | Keycloak - [not-configurable] Azure Entra ID - [configurable]                                            | Deprecated[1]              | Keycloak - [configurable] Azure Entra ID - [configurable] | Keycloak - [configurable] Azure Entra ID - [configurable] | Deprecated[2]                     |
```

Comment:
followup required with keycloak to determine if it's really impossible to configure a confidential client for authorization flows.  I'd be extremely suprrised.